### ISSUE TITLE
Option --logger-logsdir doesn't work well with pytest-xdist

### ISSUE THREAD

**Issue by aurzenligl** [aurzenligl/pytest-logger#14]
When test are run in parallel all workers race to recreate given directory (check/remove/mkdir). When second worker does this, it may remove files which first worker holds opened.

This function and it's default argument (which plugin user cannot influence) causes problem:
```
def _make_logsdir_dir(dstname, cleandir=True):
    logsdir = py.path.local(dstname)
    if cleandir:
        if logsdir.check():
            logsdir.remove()
        logsdir.mkdir()
    return logsdir
```

**Comment by andreabisello** [comment]
@aurzenligl  hi, thanks for your plugin.
i'm looking for something to move my "print" output in a file.
i was reading this https://docs.pytest.org/en/stable/logging.html?highlight=logging but then i found this plugin.
i have a massive use of pytest-xdist,
more than 2 years are passed since the creation of this bug,
the plugin is still maintained? can i use with xdist? thanks a lot.
have a nice day

**Comment by aurzenligl** [comment]
Hello @andreabisello - sorry for late answer. Plugin should work with xdist without trouble (aside from the bug regarding custom logdir, see https://pytest-logger.readthedocs.io/en/latest/usage.html#the-logs-directory-layout). It's has been used with xdist and is tested against xdist:
https://github.com/aurzenligl/pytest-logger/blob/7407d9e82cd2f79ad82e9a355434cf405998133e/tests/test_pytest_logger.py#L463

**Comment by andreabisello** [comment]
@aurzenligl  oh, thanks, so this bug is related only with the use of a customer directory.  ok i will try soon

**Comment by andreabisello** [comment]
@aurzenligl  

so pytest-logger stay on top of standard python logging and help to configure and handle log level and file.

not able to understand how

```
def pytest_logger_logdirlink(config):
    return os.path.join(os.path.dirname(__file__), 'mylogs')
```
works : mylogs folder is not created,
this is my pytest.ini

```
log_cli = True
log_cli_level = ERROR
log_cli_format = %(asctime)s %(filename)s %(funcName)s %(levelname)s %(message)s
log_cli_date_format = %d-%m-%Y %H:%M:%S
log_file = logs/log.log
log_file_level = DEBUG
log_file_format = %(asctime)s %(filename)s %(funcName)s %(levelname)s %(message)s
log_file_date_format = %d-%m-%Y %H:%M:%S
```

and this is on my conftest.py

```
def pytest_logger_config(logger_config):
    logger_config.add_loggers(['pytest_logger'], stdout_level='error', file_level='debug')
    logger_config.set_log_option_default('pytest_logger')

def pytest_logger_logdirlink(config):
    return os.path.join(os.path.dirname(__file__), 'mylogs')
```

if i remove log_file , no log file is created.

this is the top of my test file

```
import logging
pytest_logger = logging.getLogger('hub')
logging.getLogger('zeep').setLevel(logging.ERROR) 
logging.getLogger("urllib3").setLevel(logging.WARNING)
logging.getLogger("urllib3").propagate = False
```

this is how i use the log

```
def test_GetVersionHub():
    pytest_logger.error("error")
    pytest_logger.info("info")
```

this is what i obtain in the logs/log.log

```
19-08-2020 15:02:26 test_hub.py test_GetVersionHub ERROR error
19-08-2020 15:02:26 test_hub.py test_GetVersionHub INFO info
```

but only because i'm settting log_file in the pytest.ini, or i will not obtain any file.

can you kindly help me understand what i'm doing?

thanks a lot

**Comment by andreabisello** [comment]
anyway python logging with xdist don't works,
as you can read in this issue https://github.com/pytest-dev/pytest-xdist/issues/402 
but also pytest-logger don't work, it has the same problem of python logging,
using xdist the log is corrupted

log without xdist

```
19-08-2020 17:42:16 test_prima_funzionalita.py test_rosso_uno INFO info!
19-08-2020 17:42:16 test_prima_funzionalita.py test_rosso_uno WARNING warning!
19-08-2020 17:42:16 test_prima_funzionalita.py test_rosso_uno DEBUG debug!
19-08-2020 17:42:16 test_prima_funzionalita.py test_rosso_uno ERROR error!
19-08-2020 17:42:16 test_prima_funzionalita.py test_rosso_uno CRITICAL critical!
19-08-2020 17:42:16 test_prima_funzionalita.py test_rosso_due INFO info!
19-08-2020 17:42:16 test_prima_funzionalita.py test_rosso_due WARNING warning!
19-08-2020 17:42:16 test_prima_funzionalita.py test_rosso_due DEBUG debug!
19-08-2020 17:42:16 test_prima_funzionalita.py test_rosso_due ERROR error!
19-08-2020 17:42:16 test_prima_funzionalita.py test_rosso_due CRITICAL critical!
19-08-2020 17:42:16 test_prima_funzionalita.py test_rosso_tre INFO info!
19-08-2020 17:42:16 test_prima_funzionalita.py test_rosso_tre WARNING warning!
19-08-2020 17:42:16 test_prima_funzionalita.py test_rosso_tre DEBUG debug!
19-08-2020 17:42:16 test_prima_funzionalita.py test_rosso_tre ERROR error!
19-08-2020 17:42:16 test_prima_funzionalita.py test_rosso_tre CRITICAL critical!
19-08-2020 17:42:16 test_seconda_funzionalita.py test_bianco INFO info!
19-08-2020 17:42:16 test_seconda_funzionalita.py test_bianco WARNING warning!
19-08-2020 17:42:16 test_seconda_funzionalita.py test_bianco DEBUG debug!
19-08-2020 17:42:16 test_seconda_funzionalita.py test_bianco ERROR error!
19-08-2020 17:42:16 test_seconda_funzionalita.py test_bianco CRITICAL critical!
19-08-2020 17:42:16 test_seconda_funzionalita.py test_verde INFO info!
19-08-2020 17:42:16 test_seconda_funzionalita.py test_verde WARNING warning!
19-08-2020 17:42:16 test_seconda_funzionalita.py test_verde DEBUG debug!
19-08-2020 17:42:16 test_seconda_funzionalita.py test_verde ERROR error!
19-08-2020 17:42:16 test_seconda_funzionalita.py test_verde CRITICAL critical!
19-08-2020 17:42:16 test_seconda_funzionalita.py test_arancione INFO info!
19-08-2020 17:42:16 test_seconda_funzionalita.py test_arancione WARNING warning!
19-08-2020 17:42:16 test_seconda_funzionalita.py test_arancione DEBUG debug!
19-08-2020 17:42:16 test_seconda_funzionalita.py test_arancione ERROR error!
19-08-2020 17:42:16 test_seconda_funzionalita.py test_arancione CRITICAL critical!


```

log with xdist

```
19-08-2020 17:43:41 test_prima_funzionalita.py test_rosso_uno INFO info!
119-08-2020 17:43:41 test_prima_funzionalita.py test_rosso_uno WARNING warning!1919-08-2020 17:43:41 test_prima_funzionalita.py test_rosso_uno DEBUG debug19-19-08-2020 17:43:41 test_prima_funzionalita.py test_rosso_uno ERROR erro19-019-08-2020 17:43:41 test_prima_funzionalita.py test_rosso_uno CRITICAL critical!
ritical!
```

@aurzenligl  if you want i will create a repository on github to reproduce the issue

thanks

**Comment by florczakraf** [comment]
I'll only address the configuration issue.

Instead of `'pytest_logger'` in:
```
logger_config.add_loggers(['pytest_logger'], stdout_level='error', file_level='debug')
```
you should put there names of the actual loggers you'd like to capture. For example: `['hub', 'zeep', 'urllib3']`.

Reference: https://pytest-logger.readthedocs.io/en/latest/usage.html#high-level-hook

**Comment by andreabisello** [comment]
@florczakraf  @aurzenligl  i'm going off-topic in this issue, i will remove my comments and i'll move on a new issue.

**Comment by aurzenligl** [comment]
Closing due to inactivity

**Comment by aurzenligl** [comment]
Closing due to inactivity
